src/utils/ipfs/uploadDirToIpfs.ts

```diff
@@ -0,0 +1,57 @@
+import IpfsHttpClient from 'ipfs-http-client'
+import all from 'it-all'
+import path from 'path'
+const { globSource } = IpfsHttpClient
+
+const defaultIpfsApiUrl = 'http://localhost:5001'
+
+interface Cid {
+  version: number
+  codec: string
+  multihash: Buffer
+  multibaseName: string
+  toString: () => string
+}
+
+interface IpfsAddResult {
+  path: string
+  size: number
+  cid: Cid
+}
+
+/**
+ * Uploads dist folder to IPFS
+ * Applies various ignore patterns:
+ * - .ipfsignore
+ * - .gitignore
+ */
+export async function uploadDirToIpfs(
+  dirPath: string,
+  options?: {
+    ipfsApiUrl?: string
+    ignore?: string[]
+    progress?: (totalBytes: number) => void
+  }
+): Promise<string> {
+  const { ipfsApiUrl, ignore, progress } = options || {}
+  const ipfs = IpfsHttpClient(ipfsApiUrl || defaultIpfsApiUrl)
+
+  const results: IpfsAddResult[] = await all(
+    ipfs.add(globSource(dirPath, { recursive: true, ignore }), { progress })
+  )
+
+  // Warning! Infura does not allow uploading files with many files
```

Comment:
Interesting... but the API call doesn't fail?

This bit seems quite brittle but I assume we have no documentation on how Infura or other gateways would respond when this fails.

Reply:
I find out this myself while working in this PR and doing a test with a directory with many files. I haven't found a reference to this limitation but may well be out there.

Reply:
What happens with the ipfs.io API? Similar behaviour? Or it'll just accept everything?

In any case, the extra sanity check doesn't hurt, so let's leave it in!

Reply:
infura exposes an IPFS API, while ipfs.io only exposes a gateway, so I can't test against ipfs.io